e2e/LongHaul/device/IotHub.cs

```diff
@@ -158,21 +133,9 @@ public async Task RunAsync(CancellationToken ct)
             Debug.Assert(_deviceClient != null);
             Debug.Assert(telemetryObject != null);
 
-            // Save off the event time, or use "now" if not specified
-            var creationTimeUtc = telemetryObject.EventDateTimeUtc ?? DateTime.UtcNow;
-            // Remove it so it does not get serialized in the message
-            telemetryObject.EventDateTimeUtc = null;
-
-            string message = JsonSerializer.Serialize(telemetryObject, _jsonSerializerOptions);
-            Debug.Assert(!string.IsNullOrWhiteSpace(message));
-
-            var iotMessage = new Message(Encoding.UTF8.GetBytes(message))
+            var iotMessage = new TelemetryMessage(telemetryObject)
             {
-                ContentEncoding = _contentEncoding,
-                ContentType = _contentType,
```

Comment:
Does this get set by the serialization and encoding layer?

Reply:
Yes, they will be set while calling `[SendTelemetryAsync()](https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/iothub/device/src/IotHubBaseClient.cs#L153-L154)` .